javascript/javascript1/week4/homework.js

```diff
@@ -0,0 +1,188 @@
+//todo Add one or more command to your voice assistant
+// console.log(commandObj.greeting);
+const answers = [];
+const activities = [];
+let firstName = "";
+let anwerObj = {};
+let activity = "";
+
+//FIND FIRST NAME
+function findFirstName(command) {
+  firstName += command.split(" ").slice(-1).join();
+  return firstName;
+}
+
+// GREETING AND NAME
+function greetingAnswer(msg) {
+  let answer = "";
+  findFirstName(msg);
+  msg.includes("Hello my name is")
+    ? (answer += "nice to meet you" + " " + firstName)
+    : (answer +=
+        "please use correct syntax: \n Hello my name is: inser your first name here");
+  answers.push(firstName);
+  anwerObj = { msg: answer };
+  answers.push(anwerObj);
+
+  return answer;
+}
+// console.log(greetingAnswer("Hello my name is benjamin")); //nice to meet you benjamin
+
+//ACTIVITIES
+function createActivity(msg) {
+  let answer = "";
+  activity = msg.substring(msg.indexOf(" "), msg.indexOf("to my todo"));
+
+  if (msg.startsWith("Add")) {
+    activities.push(activity);
+  } else if (msg === "What is on my todo?") {
+    answer = `you have ${activities.length} todos - ${activities.join(
+      " and "
+    )}`;
+  }
+
+  return answer;
+}
+
+//REMOVE ACTIVITY
+function removeActivity(msg) {
+  let answer = "";
+
+  for (const element of activities) {
+    activity = msg.substring(msg.indexOf(" "), msg.indexOf("from my todo"));
+
+    if (element === activity) {
+      activities.splice(activities.indexOf(element), 1);
+    } else {
+      answer = `the ${activity} not found`;
+    }
+  }
+  // }
+  return answer;
+}
+
+//DATE
+function getDatOfToday() {
+  let msg = "";
+
+  const today = new Date();
+
+  const year = today.getFullYear();
+  //get month long alone
+  const monthOfToday = today.getMonth();
+  const month = (f) =>
+    Array.from(Array(12), (e, i) =>
+      new Date(25e8 * ++i).toLocaleString("en-US", { month: f })
+    );
```

Comment:
I'm not quite sure what you are attempting here but if it is to get today's month written out?

Then you could do as follows:

```
const today = new Date();
const monthIndex = today.getMonth(); // Returns the month (0-11)
const monthNames = [
  "January", "February", "March", "April", "May", "June",
  "July", "August", "September", "October", "November", "December"
];

const monthName = monthNames[monthIndex];
```